test/pieces.js

```diff
@@ -1263,4 +1263,19 @@ describe('Pieces', function() {
     assert(fs.readFileSync(path.join(__dirname, 'public', resume.attachment._url), 'utf8') === fs.readFileSync(path.join(__dirname, '/public/static-test.txt'), 'utf8'));
   });
 
+  it('should convert a piece keeping only the present fields', async () => {
+    const req = apos.task.getReq();
+
+    const productPiece = {
+      title: 'produce name',
+      color: 'red'
+    };
+
+    const converted = {};
+    await apos.modules.product.convert(req, productPiece, converted, { presentFieldsOnly: true });
+
+    assert(Object.keys(converted).length === 3);
```

Comment:
3 because there is a `copyOfId`  property always added I'm not sure if it's still used or a relic from A2.

Reply:
Yes copyOfId is fine, although it would be nice if we did not set it at all when it would be false/null/undefined.

Reply:
I removed the key if undefined.